inference-engine/src/gna_plugin/transformations/convert_padded2valid_conv.cpp

```diff
@@ -0,0 +1,542 @@
+    auto leading_transpose = ngraph::pattern::wrap_type<ngraph::opset7::Transpose>({ ngraph::pattern::any_input(), const_input },
+        ngraph::pattern::consumers_count(1));
+    auto conv = ngraph::pattern::wrap_type<ngraph::opset7::Convolution>({ leading_transpose, const_input },
+        ngraph::pattern::consumers_count(1));
+    auto trailing_transpose = ngraph::pattern::wrap_type<ngraph::opset7::Transpose>({ conv, const_input },
+        ngraph::pattern::consumers_count(1));
+    auto bias = ngraph::pattern::wrap_type<ngraph::opset7::Add>({ trailing_transpose, const_input },
+        ngraph::pattern::consumers_count(1));
+
+    ngraph::matcher_pass_callback callback = [=](ngraph::pattern::Matcher& m) {
+        const auto& pattern_map = m.get_pattern_value_map();
+        return Convert(pattern_map.at(leading_transpose).get_node_shared_ptr(), pattern_map.at(conv).get_node_shared_ptr(),
+            pattern_map.at(trailing_transpose).get_node_shared_ptr(), pattern_map.at(bias).get_node_shared_ptr(), nullptr, nullptr);
+    };
+
+    auto m = std::make_shared<ngraph::pattern::Matcher>(bias, matcher_name);
+    this->register_matcher(m, callback);
+}
+
+ConvertPaddedTransposedWithBiasAF2ValidConv::ConvertPaddedTransposedWithBiasAF2ValidConv() {
+    MATCHER_SCOPE(ConvertPaddedTransposedWithBiasAF2ValidConv);
+
+    auto const_input = ngraph::pattern::wrap_type<ngraph::opset7::Constant>();
+    auto leading_transpose = ngraph::pattern::wrap_type<ngraph::opset7::Transpose>({ ngraph::pattern::any_input(), const_input },
+        ngraph::pattern::consumers_count(1));
+    auto conv = ngraph::pattern::wrap_type<ngraph::opset7::Convolution>({ leading_transpose, const_input },
+        ngraph::pattern::consumers_count(1));
+    auto trailing_transpose = ngraph::pattern::wrap_type<ngraph::opset7::Transpose>({ conv, const_input },
+        ngraph::pattern::consumers_count(1));
+    auto bias = ngraph::pattern::wrap_type<ngraph::opset7::Add>({ trailing_transpose, const_input },
+        ngraph::pattern::consumers_count(1));
+    auto af = ngraph::pattern::wrap_type<ngraph::op::util::UnaryElementwiseArithmetic>({ bias },
+        ngraph::pattern::consumers_count(1));
+
+    ngraph::matcher_pass_callback callback = [=](ngraph::pattern::Matcher& m) {
+        const auto& pattern_map = m.get_pattern_value_map();
+        return Convert(pattern_map.at(leading_transpose).get_node_shared_ptr(), pattern_map.at(conv).get_node_shared_ptr(),
+            pattern_map.at(trailing_transpose).get_node_shared_ptr(), pattern_map.at(bias).get_node_shared_ptr(),
+            pattern_map.at(af).get_node_shared_ptr(), nullptr);
+    };
+
+    auto m = std::make_shared<ngraph::pattern::Matcher>(af, matcher_name);
+    this->register_matcher(m, callback);
+}
+
+ConvertPadded2ValidConvSet::ConvertPadded2ValidConvSet() {
+    MATCHER_SCOPE(ConvertPadded2ValidConvSet);
+
+    ngraph::pass::Manager manager(get_pass_config());
```

Comment:
BTW, how it works? ConvertPadded2ValidConvSet is MatcherPass without callback

cc @GlebKazantaev

Reply:
This code is outdated, I removed it. It didn't for 1D convolution tests, but strangely - it worked for 2D cases.